cpp/algo_module/algorithm/algo.cpp

```diff
@@ -1 +1,204 @@
 #include "algo.hpp"
+
+double Algo::toRadians(double degrees) { return degrees * M_PI / 180.0; }
+
+double Algo::haversineDistance(double lat1, double lon1, double lat2, double lon2) {
+  const double earthRadius = 6371.0;  // IN KM
+
+  lat1 = toRadians(lat1);
+  lon1 = toRadians(lon1);
+  lat2 = toRadians(lat2);
+  lon2 = toRadians(lon2);
+
+  double dLat = lat2 - lat1;
+  double dLon = lon2 - lon1;
+  double a = sin(dLat / 2) * sin(dLat / 2) + cos(lat1) * cos(lat2) * sin(dLon / 2) * sin(dLon / 2);
+  double c = 2 * atan2(sqrt(a), sqrt(1 - a));
+  double distance = earthRadius * c;
+
+  return distance;  // returns distance in km
+}
+
+/*calculates the heuristic based on haversine, or returns the value if the heuristic is custom*/
+double Algo::CalculateHeuristic(const Config &config, const mgp::Node &node, const GoalNodes &nodes) {
+  if (config.heuristic_name != "") {
```

Comment:
No no, haversine is the function that calculates the distance based on lat lon. And I left it taht way because its faster to check one if loop, rather then few of them for lat lon.

Reply:
Talked on meet